### PR TITLE
Smoke tests

### PR DESCRIPTION
**Pull request by abvgedeika**

### CODE REVIEW

File: test/smoke/dapi.js [hunk 1]
```diff
@@ -1,10 +1,58 @@
...
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const options = {
+        debug: false,
+        verbose: false,
+        errors: false,
+        warnings: false,
+        seeds: [{ ip: networkConfig.inventory._meta.hostvars[nodeName].public_ip }],
+      };
+      const dapi = new DAPIClient(options, 3000);
```
Comment (shumkov) [inline test/smoke/dapi.js]: We can't use DAPI client here :(. Please use jayson.

File: test/smoke/dapi.js [hunk 2]
```diff
@@ -1,10 +1,58 @@
...
+        expect(balance).to.be.an('number');
+        expect(balance % 1).to.be.equal(0);
+      });
 
-  // fetchDapObjects any dapiId and type
-  it('should respond data from drive');
+      // fetchDapContract any dapiId and type
+      xit('should respond fetchDapContract', async () => {
+        // TODO fetchDapContract not implemented in dashdrve
+        const dapContract = await dapi.fetchDapContract('mystring');
```
Comment (shumkov) [inline test/smoke/dapi.js]: `'mystring' -> 'fakeDapId'`?

File: test/smoke/dashCore.js [hunk 3]
```diff
@@ -1,18 +1,115 @@
...
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      it('should have correct network type', async () => {
+        const networkInfo = await rpc.getNetworkInfo();
```
Comment (shumkov) [inline test/smoke/dashCore.js]: `const { result: { networkactive, subversion } } = ...`

File: test/smoke/dashCore.js [hunk 4]
```diff
@@ -1,18 +1,115 @@
...
+        expect(networkInfo.error).to.be.equal(null);
```
Comment (shumkov) [inline test/smoke/dashCore.js]: RpcClient should throw an error if got `error` so that check doesn't make sense.

File: test/smoke/dashCore.js [hunk 5]
```diff
@@ -1,18 +1,115 @@
...
+    for (let i = 0; i < nodeNames.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeNames[i]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      const blockCount = await rpc.getBlockCount();
```
Comment (shumkov) [inline test/smoke/dashCore.js]: Why do you get blocks height from `getBlockCount` if we already have it in `getBlockchainInfo` ?

File: test/smoke/dashCore.js [hunk 6]
```diff
@@ -1,18 +1,115 @@
...
+      blockchainInfo = await rpc.getBlockchainInfo();
+      blockCounts.push(blockCount.result);
+      if (!blockhashes[blockchainInfo.result.blocks]) {
+        blockhashes[blockchainInfo.result.blocks] = blockchainInfo.result.bestblockhash;
+      }
+    }
+    expect(blockhashes[blockchainInfo.result.blocks]).to.be.equal(
```
Comment (shumkov) [inline test/smoke/dashCore.js]: This check should be in `for` loop.

File: test/smoke/dashCore.js [hunk 7]
```diff
@@ -1,18 +1,115 @@
...
+      blockchainInfo.result.bestblockhash,
+    );
+    expect(Math.max(...blockCounts) - Math.min(...blockCounts)).to.be.below(3);
```
Comment (shumkov) [inline test/smoke/dashCore.js]: We can use `Object.key(blockhashes)` instead of additional `blockCount` array

File: test/smoke/dashCore.js [hunk 8]
```diff
@@ -1,18 +1,115 @@
...
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      // masternode status
+      it('should masternodes be enabled', async () => {
+        const masternodelist = await rpc.masternodelist();
```
Comment (shumkov) [inline test/smoke/dashCore.js]: Why not just https://dash-docs.github.io/en/developer-reference#masternode--noref---status?

File: test/smoke/dashCore.js [hunk 9]
```diff
@@ -1,18 +1,115 @@
...
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[
+          networkConfig.inventory.masternodes.hosts[0]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      this.timeout(160000);
+      const blockCount = await rpc.getBlockCount();
```
Comment (shumkov) [inline test/smoke/dashCore.js]: waitfornewblock

File: test/smoke/faucet.js [hunk 10]
```diff
@@ -1,4 +1,17 @@
...
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
 describe('Faucet', () => {
-  // getFunds
-  it('should respond positive balance');
+  networkConfig.inventory.web.hosts.forEach((nodeName) => {
+    it(`should respond positive balance: ${nodeName}`,
```
Comment (shumkov) [inline test/smoke/faucet.js]: descibe(nodeName)

File: test/smoke/faucet.js [hunk 11]
```diff
@@ -1,4 +1,17 @@
...
+      async () => {
+        const faucetBalance = await fetch(`http://${networkConfig.inventory._meta.hostvars[nodeName].public_ip}/`).then(res => res.text());
+        const regex = 'Faucet balance: ([0-9,.]+)';
+        const result = faucetBalance.match(regex);
+        expect(parseFloat(parseFloat(result[1].replace(/,/g, '')))).to.be.an('number').above(0);
```
Comment (shumkov) [inline test/smoke/faucet.js]: Why double `parseFloat`? :)

File: test/smoke/insight.js [hunk 12]
```diff
@@ -1,4 +1,30 @@
...
+const networkConfig = getNetworkConfig();
+
+const fetch = require('node-fetch');
+
 describe('Insight', () => {
-  // /insight-api-dash/blocks?limit=1
-  it('should return block');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      it(`insight should return block ${nodeName}`,
```
Comment (shumkov) [inline test/smoke/insight.js]: I think `nodeName` in `describe` is enough?

File: test/smoke/insight.js [hunk 13]
```diff
@@ -1,4 +1,30 @@
...
+        async () => {
+          const response = await fetch(`http://${networkConfig.inventory._meta.hostvars[nodeName].public_ip}:3001/insight-api-dash/blocks?limit=1`);
+          const { blocks } = await response.json();
+          expect(blocks.length)
```
Comment (shumkov) [inline test/smoke/insight.js]: `expect().to.have.lengthOf()`

File: test/smoke/ipfs.js [hunk 14]
```diff
@@ -1,4 +1,16 @@
...
+const networkConfig = getNetworkConfig();
+
 describe('IPFS', () => {
   // swarm peers
-  it('should be interconnected');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    it(nodeName, async () => {
+      const ipfs = ipfsAPI(networkConfig.inventory._meta.hostvars[nodeName].public_ip, '5001', { protocol: 'http' });
+      const peers = await ipfs.swarm.peers();
+      expect(peers.length).to.be.equal(networkConfig.inventory.masternodes.hosts.length - 1);
```
Comment (shumkov) [inline test/smoke/ipfs.js]: `expect().to.have.lengthOf()`

File: test/smoke/sentinel.js [hunk 15]
```diff
@@ -1,4 +1,32 @@
+const Docker = require('dockerode');
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
 describe('Sentinel', () => {
-  // Container should be running and the log should be empty
-  it('should be running without errors');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    it(`should be running without errors ${nodeName}`, async () => {
```
Comment (shumkov) [inline test/smoke/sentinel.js]: `describe(nodeName)`

File: test/smoke/sentinel.js [hunk 16]
```diff
@@ -1,4 +1,32 @@
...
+      await dockerode.getContainer(containerId)
+        .logs({
+          stdout: true,
+          stderr: true,
+          follow: 0,
+          since: (Math.floor(new Date() / 1000) - 300), // logs from last 5 mins
+          timestamps: true,
+        })
+        .then((error, stream) => {
+          expect(error).to.be.equal('');
```
Comment (shumkov) [inline test/smoke/sentinel.js]: `expect().to.be.empty()`

File: test/smoke/sentinel.js [hunk 17]
```diff
@@ -1,4 +1,32 @@
...
+          expect(stream).to.be.equal(undefined);
```
Comment (shumkov) [inline test/smoke/sentinel.js]: `expect(stream).to.be.undefined();`

File: test/testUtils.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
+const SSH = require('simple-ssh');
```
Comment (shumkov) [inline test/testUtils.js]: We don't need utils anymore.

File: package.json [hunk 19]
```diff
@@ -30,6 +33,10 @@
     "js-yaml": "^3.12.0",
     "lodash.merge": "^4.6.1",
     "mocha": "^5.2.0",
-    "node-openvpn": "github:luigiplr/node-openvpn"
+    "node-fetch": "^2.2.0",
+    "simple-ssh": "^1.0.0",
```
Comment (shumkov) [inline package.json]: We don't need it

File: package.json [hunk 20]
```diff
@@ -21,6 +21,9 @@
   },
   "homepage": "https://github.com/dashpay/dash-network-deploy#readme",
   "devDependencies": {
+    "@dashevo/dapi-client": "git+ssh://git@github.com/dashevo/dapi-client.git#feature/new_commands",
+    "@dashevo/dashd-rpc": "git+ssh://git@github.com/dashevo/dashd-rpc.git#develop",
```
Comment (shumkov) [inline package.json]: Let's use npm package.

File: package.json [hunk 21]
```diff
@@ -21,6 +21,9 @@
   },
   "homepage": "https://github.com/dashpay/dash-network-deploy#readme",
   "devDependencies": {
+    "@dashevo/dapi-client": "git+ssh://git@github.com/dashevo/dapi-client.git#feature/new_commands",
```
Comment (shumkov) [inline package.json]: We can't use private packages.

File: test/smoke/dashCore.js (hunk 5 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: use only getBlockchainInfo now

File: test/smoke/dashCore.js (hunk 3 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: fixed

File: test/smoke/dashCore.js (hunk 4 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: removed

File: test/smoke/dapi.js (hunk 2 above)
Comment (abvgedeika) [reply test/smoke/dapi.js]: changed

File: test/smoke/dapi.js (hunk 1 above)
Comment (abvgedeika) [reply test/smoke/dapi.js]: moved on jayson

File: test/smoke/dashCore.js (hunk 6 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: done

File: test/smoke/dashCore.js (hunk 8 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: rpc masternodelist returns list of all nodes and their statuses. so, currently we verify that rpc to every node returns status "enabled" by all nodes. if you suggest to use rpc.masternode('status') we get status only by one node.
what is  the better option?

File: test/smoke/dashCore.js (hunk 9 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: changed

File: test/smoke/faucet.js (hunk 10 above)
Comment (abvgedeika) [reply test/smoke/faucet.js]: added

File: test/smoke/faucet.js (hunk 11 above)
Comment (abvgedeika) [reply test/smoke/faucet.js]: removed

File: test/smoke/insight.js (hunk 12 above)
Comment (abvgedeika) [reply test/smoke/insight.js]: removed

File: test/smoke/insight.js (hunk 13 above)
Comment (abvgedeika) [reply test/smoke/insight.js]: changed

File: test/smoke/ipfs.js (hunk 14 above)
Comment (abvgedeika) [reply test/smoke/ipfs.js]: changed

File: test/smoke/sentinel.js (hunk 15 above)
Comment (abvgedeika) [reply test/smoke/sentinel.js]: added

File: test/smoke/sentinel.js (hunk 16 above)
Comment (abvgedeika) [reply test/smoke/sentinel.js]: changed

File: test/smoke/sentinel.js (hunk 17 above)
Comment (abvgedeika) [reply test/smoke/sentinel.js]: changed

File: test/testUtils.js (hunk 18 above)
Comment (abvgedeika) [reply test/testUtils.js]: removed

File: package.json (hunk 19 above)
Comment (abvgedeika) [reply package.json]: removed

File: package.json (hunk 20 above)
Comment (abvgedeika) [reply package.json]: okay

File: package.json (hunk 21 above)
Comment (abvgedeika) [reply package.json]: removed

File: test/smoke/dapi.js [hunk 22]
```diff
@@ -1,10 +1,52 @@
...
 describe('DAPI', () => {
-  // getBlockHash height:1
-  it('should respond data from chain');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const dapiClient = jaysonClient.http({
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 3000,
+      });
+      const config = {
```
Comment (shumkov) [inline test/smoke/dapi.js]: Let's read options from `networkConfig.variables`?

File: test/smoke/dapi.js [hunk 23]
```diff
@@ -1,10 +1,52 @@
...
+      it('should respond data from insight', async () => {
+        const { result } = await dapiClient.request('getBestBlockHeight', { blockHeight: 1 });
+        expect(result).to.be.an('number');
+        expect(result % 1).to.be.equal(0);
+      });
 
-  // getBestBlockHeight
-  it('should respond data from insight');
+      // getBalance from dapi
+      it('should respond getBalance from drive', async () => {
```
Comment (shumkov) [inline test/smoke/dapi.js]: 'should respond data from insight'

File: test/smoke/dapi.js [hunk 24]
```diff
@@ -1,10 +1,52 @@
...
+        const { result } = await dapiClient.request('getBalance',
+          { address: `${networkConfig.variables.faucet_address}` });
+        expect(result).to.be.an('number');
+        expect(result % 1).to.be.equal(0);
+      });
 
-  // fetchDapObjects any dapiId and type
-  it('should respond data from drive');
+      // fetchDapContract any dapiId and type
+      xit('should respond fetchDapContract', async () => {
```
Comment (shumkov) [inline test/smoke/dapi.js]: should respond data from drive

Comment (shumkov) [reply test/smoke/dapi.js]: still doesn't work?

File: test/smoke/dashCore.js [hunk 25]
```diff
@@ -1,18 +1,107 @@
...
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+describe('All nodes', () => {
+  let nodeNames = networkConfig.inventory.masternodes.hosts;
+  nodeNames = nodeNames.concat(networkConfig.inventory['wallet-nodes'].hosts);
+  // TODO miner rpc call stuck. why?
+  // nodeNames = nodeNames.concat(networkConfig.inventory['miners'].hosts);
```
Comment (shumkov) [inline test/smoke/dashCore.js]: still stuck?

File: test/smoke/dashCore.js [hunk 26]
```diff
@@ -1,18 +1,107 @@
...
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      const { result: { blocks, bestblockhash } } = await rpc.getBlockchainInfo();
+      if (!blockhashes[blocks]) {
+        blockhashes[blocks] = bestblockhash;
+      }
+      expect(blockhashes[blocks]).to.be.equal(bestblockhash);
+    }
+    expect(Math.max(...Object.keys(blockhashes))
```
Comment (shumkov) [inline test/smoke/dashCore.js]:
```
const blocksCounts = Object.keys(blockHashes);
```

File: test/smoke/dashCore.js [hunk 27]
```diff
@@ -1,18 +1,107 @@
...
+    const blockhashes = {};
+    for (let i = 0; i < nodeNames.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeNames[i]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
```
Comment (shumkov) [inline test/smoke/dashCore.js]:
```
const blockHashes = []
```

File: test/smoke/dashCore.js (hunk 8 above)
Comment (shumkov) [reply test/smoke/dashCore.js]: Yes, probably `masternode list` check more bulletproof :)

File: test/smoke/dashCore.js (hunk 9 above)
Comment (shumkov) [reply test/smoke/dashCore.js]: We need just `await rpc.waitForNewBlock(1);` here.

File: test/smoke/insight.js [hunk 28]
```diff
@@ -1,4 +1,24 @@
...
+const networkConfig = getNetworkConfig();
+
+const fetch = require('node-fetch');
+
 describe('Insight', () => {
-  // /insight-api-dash/blocks?limit=1
-  it('should return block');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      it('insight should return block',
```
Comment (shumkov) [inline test/smoke/insight.js]: should return block

File: test/smoke/insight.js [hunk 29]
```diff
@@ -1,4 +1,24 @@
...
+        async () => {
+          const response = await fetch(`http://${networkConfig.inventory._meta.hostvars[nodeName].public_ip}:3001/insight-api-dash/blocks?limit=1`);
+          const { blocks } = await response.json();
+          expect(blocks).to.have.lengthOf(1);
+        });
+
+      it('should return master node list', async () => {
```
Comment (shumkov) [inline test/smoke/insight.js]: Why do we need this test?

File: test/smoke/sentinel.js [hunk 30]
```diff
@@ -1,4 +1,35 @@
...
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
 describe('Sentinel', () => {
-  // Container should be running and the log should be empty
-  it('should be running without errors');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      it(`should be running without errors ${nodeName}`, async () => {
```
Comment (shumkov) [inline test/smoke/sentinel.js]: should be running without errors

File: test/smoke/sentinel.js [hunk 31]
```diff
@@ -1,4 +1,35 @@
...
+        const dockerode = new Docker({
+          host: `http://${networkConfig.inventory._meta.hostvars[nodeName].public_ip}`,
+          port: 2375,
+        });
+        const opts = { filters: '{"name": ["mn_sentinel_sentinel"]}' };
+        const containers = await dockerode.listContainers(opts);
+        expect(containers.length).to.be.equal(1);
```
Comment (shumkov) [inline test/smoke/sentinel.js]: to.have.lengthOf

File: test/smoke/sentinel.js [hunk 32]
```diff
@@ -1,4 +1,35 @@
...
+        await dockerode.getContainer(containerId).logs(
+          {
+            stdout: true,
+            stderr: true,
+            follow: 0,
+            since: (Math.floor(new Date() / 1000) - 300), // logs from last 5 mins
+            timestamps: true,
+          },
+        )
+          .then((error, stream) => {
```
Comment (shumkov) [inline test/smoke/sentinel.js]: why do you need `then`? 

const stream = await dockerode.getContainer(containerId).logs(...)
expect(stream).to.be.undefined();

File: test/testUtils_js [hunk 33]
```diff
@@ -0,0 +1,28 @@
+const SSH = require('simple-ssh');
```
Comment (shumkov) [inline test/testUtils_js]: let's remove it

File: test/smoke/dapi.js (hunk 22 above)
Comment (abaranouski) [reply test/smoke/dapi.js]: changed

File: test/smoke/dapi.js (hunk 23 above)
Comment (abaranouski) [reply test/smoke/dapi.js]: changed

File: test/smoke/dapi.js (hunk 24 above)
Comment (abaranouski) [reply test/smoke/dapi.js]: { jsonrpc: '2.0',
  id: 'fc886477-375b-4752-b14c-164143bb1b5f',
  error: { code: 100, message: 'Initial sync in progress' } }

File: test/smoke/dashCore.js (hunk 9 above)
Comment (abaranouski) [reply test/smoke/dashCore.js]: await rpc.waitForNewBlock(1) doesn't work now.
for example the following code
``` 
     const { result } = await rpc.getBlockCount();
      const { result: { height } } = await rpc.waitForNewBlock(1);
      expect(result + 1).to.be.equal(height);
```
doesn't wait new block. and rpc.waitForNewBlock returns height=result imediately

File: test/smoke/dashCore.js (hunk 25 above)
Comment (abaranouski) [reply test/smoke/dashCore.js]: await rpc.waitForNewBlock(1) doesn't work now.
for example the following code

     const { result } = await rpc.getBlockCount();
      const { result: { height } } = await rpc.waitForNewBlock(1);
      expect(result + 1).to.be.equal(height);
doesn't wait new block. and rpc.waitForNewBlock returns height=result immediately

File: test/smoke/dashCore.js (hunk 26 above)
Comment (abaranouski) [reply test/smoke/dashCore.js]: changed

File: test/smoke/dashCore.js (hunk 27 above)
Comment (abaranouski) [reply test/smoke/dashCore.js]: do you mean blockhashes -> blockHashes ? or what?

File: test/smoke/insight.js (hunk 28 above)
Comment (abaranouski) [reply test/smoke/insight.js]: done

File: test/smoke/insight.js (hunk 29 above)
Comment (abaranouski) [reply test/smoke/insight.js]: why not? isight should sync data from all master nodes. do you think it's extra verification? I can remove it. btw, now it failed due to AssertionError: expected [] to have a length of 13 but got 0

File: test/smoke/sentinel.js (hunk 30 above)
Comment (abaranouski) [reply test/smoke/sentinel.js]: done

File: test/smoke/sentinel.js (hunk 31 above)
Comment (abaranouski) [reply test/smoke/sentinel.js]: done

File: test/smoke/sentinel.js (hunk 32 above)
Comment (abaranouski) [reply test/smoke/sentinel.js]: removed

File: test/testUtils_js (hunk 33 above)
Comment (abaranouski) [reply test/testUtils_js]: ah, committed by mistake

File: test/smoke/dapi.js (hunk 22 above)
Comment (shumkov) [reply test/smoke/dapi.js]: port?

File: test/smoke/dashCore.js (hunk 27 above)
Comment (shumkov) [reply test/smoke/dashCore.js]: Yep. And you should define array before you use it.

File: test/smoke/insight.js (hunk 29 above)
Comment (shumkov) [reply test/smoke/insight.js]: 'Coz here we just test that insight is working. We don't test that masternode list working properly in dash core

File: test/smoke/faucet.js [hunk 34]
```diff
@@ -1,4 +1,20 @@
...
+
 describe('Faucet', () => {
-  // getFunds
-  it('should respond positive balance');
+  networkConfig.inventory.web.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      it('should respond positive balance',
+        async () => {
+          const faucetBalance = await fetch(`http://${networkConfig.inventory._meta.hostvars[nodeName].public_ip}/`)
+            .then(res => res.text());
```
Comment (shumkov) [inline test/smoke/faucet.js]: Please avoid of mixing async/await and promises approach.

const response = await fetch()
const balance = await response.text()

File: test/smoke/dapi.js [hunk 35]
```diff
@@ -1,10 +1,52 @@
...
+      it('should respond data from insight', async () => {
+        const { result } = await dapiClient.request('getBestBlockHeight', { blockHeight: 1 });
+        expect(result).to.be.an('number');
+        expect(result % 1).to.be.equal(0);
+      });
 
-  // getBestBlockHeight
-  it('should respond data from insight');
+      // getBalance from dapi
+      it('should respond data from insight', async () => {
```
Comment (shumkov) [inline test/smoke/dapi.js]: Why do we have two duplicate tests/

File: test/smoke/insight.js (hunk 29 above)
Comment (abvgedeika) [reply test/smoke/insight.js]: removed the test completely

File: test/smoke/faucet.js (hunk 34 above)
Comment (abvgedeika) [reply test/smoke/faucet.js]: modified

File: test/smoke/dapi.js (hunk 35 above)
Comment (abvgedeika) [reply test/smoke/dapi.js]: will remove "getBalance from dapi". getBestBlockHeight remains

File: test/smoke/dashCore.js [hunk 36]
```diff
@@ -101,7 +101,7 @@ describe('Masternodes', () => {
       const rpc = new RpcClient(config);
       this.timeout(160000);
```
[the remark this answers is not in the document]
Comment (abvgedeika) [reply test/smoke/dashCore.js]: hm, I thought block time is 2.5 min for dash, no?

File: test/smoke/dashCore.js [hunk 37]
```diff
@@ -1,18 +1,106 @@
...
+        user: networkConfig.variables.dashd_rpc_user,
+        pass: networkConfig.variables.dashd_rpc_password,
+        host: networkConfig.inventory._meta.hostvars[
+          networkConfig.inventory.masternodes.hosts[0]].public_ip,
+        port: networkConfig.variables.dashd_rpc_port,
+      };
+      const rpc = new RpcClient(config);
+      const { result } = await rpc.getBlockCount();
+      const { result: { height } } = await rpc.waitForNewBlock(150000);
+      expect(result + 1).to.be.equal(height);
```
Comment (shumkov) [inline test/smoke/dashCore.js]: This check is unnecessary. `waitForNewBlock` timeout should be more than test timeout.

File: test/smoke/ipfs.js [hunk 38]
```diff
@@ -1,4 +1,16 @@
...
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
 describe('IPFS', () => {
   // swarm peers
-  it('should be interconnected');
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    it(nodeName, async () => {
```
Comment (shumkov) [inline test/smoke/ipfs.js]:
```
descibe(nodeName)
it ('should be interconnected')
```

File: test/smoke/dapi.js [hunk 39]
```diff
@@ -1,10 +1,44 @@
...
+        const { result } = await dapiClient.request('getBestBlockHeight', { blockHeight: 1 });
+        expect(result).to.be.an('number');
+        expect(result % 1).to.be.equal(0);
+      });
 
-  // fetchDapObjects any dapiId and type
-  it('should respond data from drive');
+      // fetchDapContract any dapiId and type
+      xit('should respond data from drive', async () => {
```
[the remark this answers is not in the document]
Comment (abvgedeika) [reply test/smoke/dapi.js]: currently we get:
{ jsonrpc: '2.0',
  id: 'd433121a-f9b8-4388-aa39-c8ee2690c223',
  error: { code: 100, message: 'Initial sync in progress' } }
so, I guess the test should be still skipped but added the following verification code
```
       const { result: { error: { code } } } = await dapiClient.request('fetchDapContract', { dapId: 'fakeDapId' });
        // we expect code -32602( invalid dap id)
        expect(code).to.be.equals(-32602);
```

File: test/smoke/dashCore.js [hunk 40]
```diff
@@ -1,18 +1,107 @@
...
+        }
+        expect(masterIps.sort()).to.deep.equal(masterIpsInv.sort());
+      });
+    });
+  });
+
   describe('Miners', () => {
     // waitfornewblock
-    it('should mine blocks');
+    it('should mine blocks', async function () {
```
[the remark this answers is not in the document]
Comment (abvgedeika) [reply test/smoke/dashCore.js]: added

File: test/smoke/dashCore.js (hunk 37 above)
Comment (abvgedeika) [reply test/smoke/dashCore.js]: waitForNewBlock always returns _current block if new block was not mined during wait time. we should leave this verification

File: test/smoke/ipfs.js (hunk 38 above)
Comment (abvgedeika) [reply test/smoke/ipfs.js]: added

File: test/smoke/dapi.js (hunk 39 above)
[the remark this answers is not in the document]
Comment (abvgedeika) [reply test/smoke/dapi.js]: disagree here: initial sync could be potential issue for integration verification and now it's so. current environment is broken. we are not able to work with dapi/drive. but if you insist I will add 100 as a positive option.

File: test/smoke/dashCore.js (hunk 37 above)
Comment (shumkov) [reply test/smoke/dashCore.js]: There are two valid approaches:
1. Wait for a new block with timeout bigger than test timeout.
2. Wait for new block with timeout less than test timeout and then check block height.

I talked about the first one. You implemented the second one. Let's keep your version but please add change assertion to make test more descriptive:
```
expect(height).to.be.above(previousHeight, 'no new blocks');
```

Comment (abvgedeika) [reply test/smoke/dashCore.js]: if wait for a new block with timeout bigger than test timeout and no new block test stuck. test timeout should be bigger then waitForNewBlock timeout

Comment (shumkov) [reply test/smoke/dashCore.js]: Ahhh... because of `await`, right.

**Pull request merged**